### PR TITLE
Add compression example

### PR DESCRIPTION
**Pull request by AspirinSJL**
Based on the hello world example.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,870      Total (=)      2,020,870

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,177,122      Total (=)     11,177,122

 No significant differences in binary sizes


```

File: examples/cpp/compression/greeter_client.cc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    // Container for the data we expect from the server.
+    HelloReply reply;
+
+    // Context for the client. It could be used to convey extra information to
+    // the server and/or tweak certain RPC behaviors.
+    ClientContext context;
+
+    // Set the client's compression algorithm to GZIP.
+    context.set_compression_algorithm(GRPC_COMPRESS_GZIP);
```
Comment (hcaseyal) [inline examples/cpp/compression/greeter_client.cc]: Is there a way for the user to verify what compression algorithm has been set on client or server side?

Comment (AspirinSJL) [reply examples/cpp/compression/greeter_client.cc]: I guess not. @muxi said the verification is hidden in some private API. The compression work is kind of transparent to the users.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,870      Total (=)      2,020,870

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,177,121      Total (<)     11,177,127

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,870      Total (=)      2,020,870

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,177,124      Total (<)     11,177,129

 No significant differences in binary sizes


```

**Pull request merged**